ecstest/testcases/dataplane/object_list_test_extend.py

```diff
@@ -0,0 +1,110 @@
+# __CR__
+# Copyright (c) 2008-2015 EMC Corporation
+# All Rights Reserved
+#
+# This software contains the intellectual property of EMC Corporation
+# or is licensed to EMC Corporation from third parties.  Use of this
+# software and the intellectual property contained therein is expressly
+# limited to the terms and conditions of the License Agreement under which
+# it is provided by or on behalf of EMC.
+# __CR__
+
+'''
+Author: Rubicon ISE team
+'''
+
+from boto.s3.bucket import Bucket
+from nose.plugins.attrib import attr
+from nose.tools import eq_ as eq
+
+from ecstest import tag, testbase, bucketname
+from ecstest.logger import logger
+from ecstest import utils
+from ecstest.dec import triage
+from ecstest.dec import not_supported
+
+
+@attr(tags=[tag.DATA_PLANE, tag.OBJECT_IO])
+class TestObjectList(testbase.EcsDataPlaneTestBase):
+    """
+    Post several objects and list them within a bucket
+    """
+    def setUp(self):
+        super(TestObjectList, self).setUp(create_bucket=True)
+
+    def tearDown(self):
+        super(TestObjectList, self).tearDown()
+
+    def _create_bucket(self, bucket_name=None):
+        """
+        To create bucket with bucket_name
+        """
+        if bucket_name is None:
+            bucket_name = bucketname.get_unique_bucket_name()
+
+        logger.debug("Create bucket: %s", bucket_name)
+        bucket = self.data_conn.create_bucket(bucket_name)
+        eq(isinstance(bucket, Bucket), True)
+
+        return bucket
+
+    def _create_keys(self, keys=[]):
+        """
+        Populate a (specified or new) bucket with objects with
+        specified names (and contents identical to their names).
+        """
+        for s in keys:
+            key = self.bucket.new_key(s)
+            key.set_contents_from_string(s)
+
+    @triage
+    # corresponding tests: test_bucket_list_distinct().
+    def test_object_list_from_distinct_bucket(self):
```

Comment:
add a comment to indicate which test case (https://github.com/ceph/s3-tests/blob/master/s3tests/functional/test_s3.py)  was ported from.